test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresInCompilerGeneratedCode.cs

```diff
@@ -1567,6 +1603,54 @@ static IEnumerable<int> TestDynamicallyAccessedMethodViaGenericMethodParameterIn
 				yield return 0;
 			}
 
+			[ExpectedWarning ("IL2026", "--IteratorLocalFunction--")]
+			[ExpectedWarning ("IL3002", "--IteratorLocalFunction--", ProducedBy = ProducedBy.Analyzer)]
+			[ExpectedWarning ("IL3050", "--IteratorLocalFunction--", ProducedBy = ProducedBy.Analyzer)]
+			static void TestLocalFunctionInIteratorLocalFunction ()
+			{
+				IteratorLocalFunction ();
+
+				[RequiresUnreferencedCode ("--IteratorLocalFunction--")]
+				[RequiresAssemblyFiles ("--IteratorLocalFunction--")]
+				[RequiresDynamicCode ("--IteratorLocalFunction--")]
+				IEnumerable<int> IteratorLocalFunction ()
+				{
+					yield return 0;
+					LocalFunction ();
+					MethodWithRequires ();
+					yield return 1;
+
+					// Trimmer doesn't try to associate LocalFunction with IteratorLocalFunction
+					[ExpectedWarning ("IL2026", "--MethodWithRequires--", ProducedBy = ProducedBy.Trimmer)]
+					void LocalFunction () => MethodWithRequires ();
```

Comment:
Should we create an issue for this? More as a documentation.
In theory this should be fixable, I would expect the local function to be generated on the state machine type, right?

Reply:
No, the local function actually gets generated outside the state machine type so I think it's similar to the case of non-state-machine nested functions (this one https://github.com/dotnet/linker/pull/2689#discussion_r838799632). And this seems to be the case even if it captures state from the state machine - all of the captured state is hoisted into a separate type next to the state machine. Example: https://sharplab.io/#v2:CYLg1APgAgTAjFADAAinALAbgLACg0Bsq6yAsgBQCUyA3nsg8vY2gMwA8AlgHYAuAfGSq1mjRqLHIevKcgC8yRDlySWJADIAxZMLmDOYMMtUM0cVAHZFxxgF88toA===

Reply:
I filed https://github.com/dotnet/linker/issues/2727, please feel free to add to it if I am missing anything specific to state machines.